File: Tokopedia/Classses/Object/Shops/Notes/NotesResult.m
//
//  NotesResult.m
//  Tokopedia
//
//  Created by IT Tkpd on 10/10/14.
//  Copyright (c) 2014 TOKOPEDIA. All rights reserved.
//

#import "NotesResult.h"

@implementation NotesResult
+(RKObjectMapping *)mapping{
    RKObjectMapping *resultMapping = [RKObjectMapping mappingForClass:[NotesResult class]];
    RKRelationshipMapping *listRel = [RKRelationshipMapping relationshipMappingFromKeyPath:kTKPD_APILISTKEY toKeyPath:kTKPD_APILISTKEY withMapping:[NotesList mapping]];
    [resultMapping addPropertyMapping:listRel];
    return resultMapping;
}
@end

File: Tokopedia/Classses/Object/Shops/Notes/NotesResult.swift
//
//  NotesResult.swift
//  Tokopedia
//
//  Created by Tedo Pranowo on 7/1/16.
//  Copyright © 2016 TOKOPEDIA. All rights reserved.
//

import Foundation
import RestKit

class NotesResult : NSObject{
    
    var is_allow : NSString = ""
    var allow_add : NSString = ""
    var has_terms : NSString = ""
    var list : Array<NotesList> = Array<NotesList>()
    var detail : NoteDetails = NoteDetails()
    
    class fileprivate func attributeMappingDictionary() -> [AnyHashable: Any]! {
        return ["is_allow" : "is_allow",
                "allow_add" : "allow_add",
                "has_terms" : "has_terms"]
    }
    
    class func mapping() -> RKObjectMapping! {
        let mapping : RKObjectMapping = RKObjectMapping(for: self)
        
        mapping.addAttributeMappings(from:self.attributeMappingDictionary())
        
        let listMapping = RKRelationshipMapping(fromKeyPath: "list", toKeyPath: "list", with: NotesList.mapping())
        mapping.addPropertyMapping(listMapping)
        
        return mapping
    }
}
